internal/provider/adc/config.go

```diff
@@ -86,24 +87,32 @@ func (d *adcClient) getConfigsForGatewayProxy(tctx *provider.TranslateContext, g
 		}
 		_, ok := tctx.Services[namespacedName]
 		if !ok {
-			return nil, errors.New("no service found for service reference")
+			return nil, errors.Errorf("no service found for service reference: %s", namespacedName)
 		}
-		endpoint := tctx.EndpointSlices[namespacedName]
-		if endpoint == nil {
-			return nil, nil
-		}
-		upstreamNodes, err := d.translator.TranslateBackendRef(tctx, v1.BackendRef{
-			BackendObjectReference: v1.BackendObjectReference{
-				Name: v1.ObjectName(provider.ControlPlane.Service.Name),
-				Port: ptr.To(v1.PortNumber(provider.ControlPlane.Service.Port)),
-			},
-		})
-		if err != nil {
-			return nil, err
-		}
-		for _, node := range upstreamNodes {
-			config.ServerAddrs = append(config.ServerAddrs, fmt.Sprintf("http://%s:%d", node.Host, node.Port))
+
+		var (
+			endpointSlices = tctx.EndpointSlices[namespacedName]
+			port           = strconv.FormatInt(int64(provider.ControlPlane.Service.Port), 10)
```

Comment:
The service port is not always equal to the container port.

Reply:
We need to match endpoints ports based on the targetPort or portName of the service.